packages/story-editor/src/components/footer/pagepreview/index.js

```diff
@@ -97,36 +97,51 @@ const Image = styled.img`
 `;
 
 // PagePreview is used in the editor's Carousel as well as in the Checklist and GridView
-function PagePreview({ page, label, ...props }) {
+function PagePreview({
+  page,
+  label,
+  isCacheable = false,
+  cachedImage = null,
+  setCachedImage = null,
+  ...props
+}) {
   const { backgroundColor } = page;
   const { width, height, isActive } = props;
 
-  const [imageBlob, setImageBlob] = useState();
   const [pageNode, setPageNode] = useState();
   const setPageRef = useCallback((node) => node && setPageNode(node), []);
   const pageAtGenerationTime = useRef();
-  const enableThumbnailCaching = useFeature('enableThumbnailCaching');
+  const enableThumbnailCaching =
+    useFeature('enableThumbnailCaching') && isCacheable;
 
   // Whenever the page is re-generated
   // remove the old (and now stale) image blob
   useEffect(() => {
-    if (pageAtGenerationTime.current !== page) {
-      setImageBlob(null);
+    if (
+      enableThumbnailCaching &&
+      isActive &&
+      pageAtGenerationTime.current !== page
+    ) {
+      setCachedImage({ pageId: page.id, cachedImage: null });
       pageAtGenerationTime.current = null;
     }
-  }, [page]);
+  }, [page, setCachedImage, isActive, enableThumbnailCaching]);
```

Comment:
This is a bit ugly. Basically, I had to add `isActive` here because otherwise when the carousel remounted, all the pages would report themselves as being "new", because they basically are - they all forgot which page they snapshotted, so they all just think they're have to re-snapshot, which would re-introduce the problem I was trying to fix anyway.

This particular "fix" can lead to stale images being shown in the carousel because if you:

* close the carousel
* switch page
' make some changes
* switch page again
* re-expand the carousel

The now inactive page, that changed while the carousel was collapsed, will be showing the wrong (old) thumbnail.

This is for sure a bug but solving it requires moving more of the snapshot logic to the carousel provider, which is only a temporary location for the caching logic anyway, as all the logic has to go further up the stack to be useful throughout the application. I did not want to go that far in this PR, so decided this was an acceptable middle-ground.

The above bug is probably pretty rare to see and will fix itself as soon as you switch to the page with the wrong thumbnail to make it the active page.

I felt this was a reasonable "bug" to leave in to improve the rendering without doing too much work, which would have to be re-done anyway at a later point when we implement the proper long-term fix for the image cache.